addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -111,6 +111,12 @@ options.registry.TableOfContent = options.Class.extend({
         const exception = (tocEl) => !tocEl.querySelector('.s_table_of_content_navbar a');
         this._activateScrollSpy(exception);
         this.oldHeadingsEls = [...headingsEls.map(el => el.cloneNode(true))];
+        const blockTextContent = this.$target[0].textContent.replace(/\s*\n\s*/g, '');
+        if (targetedEls.length === 0 && blockTextContent === '') {
```

Comment:
- This won't work if there are no more h1 h2 and you then delete the last text in the block.

- Also, I think it would be easier for future developers to understand the code if you used replaceAll() instead of a regex.

Reply:
>This won't work if there are no more h1 h2 and you then delete the last text in the block.

I don't know when it's going to be a real-life case where we have a ToC without having `h1`s.? I will check what we can do here.

Reply:
> This won't work if there are no more h1 h2 and you then delete the last text in the block

What was happening here is that when we removed the heading elements, and then made any changes in the DOM we got `areHeadingsEqual => return` so we never reached `remove_snippet`.